src/controller/ReadInteraction.h

```diff
@@ -64,13 +65,23 @@ CHIP_ERROR ReportAttribute(Messaging::ExchangeManager * exchangeMgr, EndpointId
     auto readClient =
         chip::Platform::MakeUnique<app::ReadClient>(engine, exchangeMgr, callback->GetBufferedCallback(), readParams.mReportType);
 
-    ReturnErrorOnFailure(readClient->SendRequest(readParams));
+    if (readClient->IsSubscriptionType())
+    {
+        err = readClient->SendAutoResubscribeRequest(std::move(readParams));
+    }
+    else
+    {
+        err = readClient->SendRequest(readParams);
+        delete readParams.mpAttributePathParamsList;
```

Comment:
But the unique_ptr will also delete it, no?  Isn't this a double-delete?  (Ignoring for the moment the wrong-allocator problem.)

Reply:
yes, just removed it, thanks